packages/react/src/Trans.tsx

```diff
@@ -18,9 +21,10 @@ export type TransProps = {
   components?: { [key: string]: React.ElementType | any }
   formats?: Record<string, unknown>
   children?: React.ReactNode
+} & MaximumOneOf<{
   component?: React.ComponentType<TransRenderProps>
   render?: (props: TransRenderProps) => React.ReactElement<any, any> | null
-}
+}>
```

Comment:
interesting, more simple form will not work? 

```ts
export type TransProps = { [...] } 
   & (
       {component?: React.ComponentType<TransRenderProps>} 
       |  { render?: (props: TransRenderProps) => React.ReactElement<any, any> | null}
    )
```

Reply:
tried that first, and it didn't work as well